doc/user/flowspec.rst

```diff
@@ -0,0 +1,345 @@
+
+Below example is an illustration of what Flowspec will inject in the underlying
+system:
+
+.. code-block:: shell
+
+   # linux shell
+   ipset create match0x102 hash:net,net counters
+   ipset add match0x102 32.0.0.0/16,40.0.0.0/16
+   iptables -N match0x102 -t mangle
+   iptables -A match0x102 -t mangle -j MARK --set-mark 102
+   iptables -A match0x102 -t mangle -j ACCEPT
+   iptables -i ntfp3 -t mangle -I PREROUTING -m set --match-set match0x102 src,dst
+                -g match0x102
+   ip rule add fwmark 102 lookup 102
+   ip route add 40.0.0.0/16 via 44.0.0.2 table 102
+
+For handling an incoming Flowspec entry, the following workflow is applied:
+
+- incoming Flowspec entries are handled by *bgpd*, and are stored in the BGP RIB.
+- Flowspec entry is installed according to its complexity.
+
+It will be installed if one of the following filtering action is seen on the BGP
+extended community: either redirect IP, or redirect VRF, in conjunction with rate
+option, for redirecting traffic. Or rate option set to 0, for discarding traffic.
+
+According to the degree of complexity of the Flowspec entry, it will be installed
+in *zebra* RIB. For more information about what is supported in the FRR implementation
+as rule, see :ref:`flowspec-known-issues` chapter. Flowspec entry is split in several
+parts before being sent to *zebra*.
+
+- *zebra* daemon receives the policy routing configuration
+
+Policy Based Routing entities necessary to policy route the traffic in the underlying system,
+are received by *zebra*. Two filtering contexts will be created or appended in ``Netfilter``:
+``ipset`` and ``iptable`` context. The former is used to define an IP filter based on
+multiple criterium. For instance, an ipset ``net:net`` is based on two ip addresses, while 
+``net,port,net`` is based on two ip addresses and one port ( for ICMP, UDP, or TCP). The way
+the filtering is used ( for example, is src port or dst port used ?) is defined by the
+latter filtering context. ``iptable`` command will reference the ``ipset`` context and will
+tell how to filter and what to do. In our case, a marker will be set to indicate ``iproute2`
+where to forward the traffic to. Sometimes, for dropping action, there is no need to add a
+marker; the ``iptable`` will tell to drop all packets matching the ``ipset`` entry. 
+
+Configuration guide
+-------------------
+
+In order to configure an IPv4 Flowspec engine, use the following configuration. As of
+today, it is only possible to configure Flowspec on the default VRF.
+
+.. code-block:: frr
+
+   router bgp <AS>
+     neighbor <A.B.C.D> remote-as <remoteAS>
+     address-family ipv4 flowspec
+      neighbor <A.B.C.D> activate
+    exit
+   exit
+
+You can see Flowspec entries, by using one of the following show commands:
+
+.. index:: show bgp ipv4 flowspec [detail | A.B.C.D]
+.. clicmd:: show bgp ipv4 flowspec [detail | A.B.C.D]
+
+
+Per-Interface Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Please use `^` for this. http://frrouting.readthedocs.io/projects/dev-guide/en/latest/workflow.html#documentation

Reply:
ok. subsection.